url8e6440c8aa_robdimsdale_project_euler_tgz-pJ8-com_rmd_personal_projecteuler_MainJ8/src/com/rmd/personal/projecteuler/p20to39/P033.java

```diff
@@ -0,0 +1,136 @@
+package com.rmd.personal.projecteuler.p20to39;
+
+import com.rmd.personal.projecteuler.Common;
+import com.rmd.personal.projecteuler.Problem;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class P033 implements Problem {
+
+    private static final int DEFAULT_START = 11;
+    private static final int DEFAULT_NUMERATOR_END = 98;
+    private static final int DEFAULT_DENOMINATOR_END = 99;
+
+    private int numeratorStart;
+    private int numeratorEnd;
+    private int denominatorEnd;
+
+    public P033() {
+        this.setNumeratorStart(DEFAULT_START);
+        this.setNumeratorEnd(DEFAULT_NUMERATOR_END);
+        this.setDenominatorEnd(DEFAULT_DENOMINATOR_END);
+    }
+
+    public int getNumeratorStart() {
+        return numeratorStart;
+    }
+
+    public void setNumeratorStart(int numeratorStart) {
+        this.numeratorStart = numeratorStart;
+    }
+
+    public int getNumeratorEnd() {
+        return numeratorEnd;
+    }
+
+    public void setNumeratorEnd(int start) {
+        this.numeratorEnd = start;
+    }
+
+    public int getDenominatorEnd() {
+        return denominatorEnd;
+    }
+
+    public void setDenominatorEnd(int denominatorEnd) {
+        this.denominatorEnd = denominatorEnd;
+    }
+
+    @Override
+    public String getDescription() {
+        return "The fraction 49/98 is a curious fraction, as an inexperienced mathematician in attempting to "
+                + "simplify it may incorrectly believe that 49/98 = 4/8, which is correct, is obtained by cancelling "
+                + "the 9s.\n\n"
+                + "We shall consider fractions like, 30/50 = 3/5, to be trivial examples.\n\n"
+                + "There are exactly four non-trivial examples of this type of fraction, less than one in value, and "
+                + "containing two digits in the numerator and denominator.\n\n"
+                + "If the product of these four fractions is given in its lowest common terms, "
+                + "find the value of the denominator.";
+    }
+
+    @Override
+    public String run() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 66-80 into a separate method.